egs/sprakbanken_swe/s5/local/sprakparser.py

```diff
@@ -0,0 +1,188 @@
+            if "Speaker" in line:         
+                self.speaker_id = self.get_speaker_vars(line).strip()
+            elif "Name" in line:
+                self.name = self.get_speaker_vars(line)
+            elif "Age" in line:
+                self.age = self.get_speaker_vars(line)
+            elif "Sex" in line:
+                self.gender = self.get_speaker_vars(line)
+            elif "Youth" in line:
+                self.youth = self.get_speaker_vars(line)
+            elif "Dialect" in line:
+                self.dialect = self.get_speaker_vars(line)
+            elif line == "":
+                return 
+            else:
+                pass
+
+    def print_speaker_info(self, path):
+        filename = self.filestem+ ".speaker_info"
+        dest = os.path.join(path, filename)
+        fout = codecs.open(dest, "w", "utf8")
+        fout.write("Original .spl file:\t" +self.source)
+        fout.write("\nID:\t" +self.speaker_id)
+        fout.write("\nName:\t" +self.name)
+        fout.write("\nAge:\t" +self.age)
+        fout.write("\nGender:\t" +self.gender)
+        fout.write("\nDialect:\t" +self.dialect)
+        fout.write("\nOrigin:\t" +self.youth)
+        fout.close()
+
+            
+    def set_path_vars(self, topfolder, handle):
+        for input in handle:
+            #print("path")
+            
+            line = input.strip()
+            if "recordings" in line:
+                self.recordings = self.get_vars(line)
+            elif "Directory" in line:
+                self.splpath = self.get_vars(line)[3:].replace("\\", "/") # removes "c:\"
+                
+                self.wavdir = self.wavpath(topfolder)
+            elif line == "":
+                return 
+            else:
+                pass
+            
+    def set_channel_vars(self, handle):
+        for input in handle:
+            line = input.strip()
+            if line == "":
+                return
+            else:
+                pass
+            
+    def create_filename(self, uid, file_ending):
+        return self.filestem+ "." +self.speaker_id+ "." +str(uid)+ "." +file_ending
+        
+    def wavpath(self, topfolder):
+        prefix, suffix = topfolder.rsplit('/data/', 1)
+        testpath = os.path.join(prefix, 'speech', suffix)
+        #testpath = topfolder.replace("data", "speech")
+        if os.path.exists(testpath):
+            return os.path.join(testpath, self.filestem)
+        else:
+            testpath = os.path.join(prefix, 'Speech', suffix)
+            return os.path.join(testpath, self.filestem)
+            
+    def get_vars(self, line):
+        return line.split("=")[-1]
+    
+    def get_speaker_vars(self, line) :
+        vals = self.get_vars(line).split(self.delimit)
+        return vals[-2]
+
+        
+
+
+
+if __name__ == '__main__':
+    
+    s = Session("/home/andreas/Desktop/aligned_corpus/da.16kHz.0565-2", 
+                "adb_0565/data/scr0565/06/05650606/r5650555.spl")
+    
```

Comment:
I guess as a self test this is ok -- but you could allow to have the file specified in argv and the hardcoded path would be used only as a fallback